src/main/java/jcifs/smb1/smb1/SmbFileOutputStream.java

```diff
@@ -0,0 +1,261 @@
+ * second argument is <code>true</code>, then bytes will be written to the
+ * end of the file rather than the beginning.
+ *
+ * @param url An smb URL string representing the file to write to
+ * @param append Append to the end of file
+ */
+
+    public SmbFileOutputStream( String url, boolean append ) throws SmbException, MalformedURLException, UnknownHostException {
+        this( new SmbFile( url ), append );
+    }
+
+/**
+ * Creates an {@link java.io.OutputStream} for writing bytes to a file
+ * on an SMB server addressed by the <code>SmbFile</code> parameter. See
+ * {@link jcifs.smb1.smb1.SmbFile} for a detailed description and examples of
+ * the smb URL syntax. If the second argument is <code>true</code>, then
+ * bytes will be written to the end of the file rather than the beginning.
+ * 
+ * @param file An <code>SmbFile</code> representing the file to write to
+ * @param append Append to the end of file
+ */
+
+    public SmbFileOutputStream( SmbFile file, boolean append ) throws SmbException, MalformedURLException, UnknownHostException {
+        this( file, append, append ? SmbFile.O_CREAT | SmbFile.O_WRONLY | SmbFile.O_APPEND :
+                                    SmbFile.O_CREAT | SmbFile.O_WRONLY | SmbFile.O_TRUNC );
+    }
+/**
+ * Creates an {@link java.io.OutputStream} for writing bytes to a file
+ * on an SMB server addressed by the <code>SmbFile</code> parameter. See
+ * {@link jcifs.smb1.smb1.SmbFile} for a detailed description and examples of
+ * the smb URL syntax.
+<p>
+The second parameter specifies how the file should be shared. If
+<code>SmbFile.FILE_NO_SHARE</code> is specified the client will
+have exclusive access to the file. An additional open command
+from jCIFS or another application will fail with the "file is being
+accessed by another process" error. The <code>FILE_SHARE_READ</code>,
+<code>FILE_SHARE_WRITE</code>, and <code>FILE_SHARE_DELETE</code> may be
+combined with the bitwise OR '|' to specify that other peocesses may read,
+write, and/or delete the file while the jCIFS user has the file open.
+ * 
+ * @param url An smb URL representing the file to write to
+ * @param shareAccess File sharing flag: <code>SmbFile.FILE_NOSHARE</code> or any combination of <code>SmbFile.FILE_READ</code>, <code>SmbFile.FILE_WRITE</code>, and <code>SmbFile.FILE_DELETE</code>
+ */
+
+    public SmbFileOutputStream( String url, int shareAccess ) throws SmbException, MalformedURLException, UnknownHostException {
+        this( new SmbFile( url, "", null, shareAccess ), false );
+    }
+
+    SmbFileOutputStream( SmbFile file, boolean append, int openFlags ) throws SmbException, MalformedURLException, UnknownHostException {
+        this.file = file;
+        this.append = append;
+        this.openFlags = openFlags;
+        this.access = (openFlags >>> 16) & 0xFFFF;
+        if( append ) {
+            try {
+                fp = file.length();
+            } catch( SmbAuthException sae ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)